vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/model/implementation/Serializer.java

```diff
@@ -0,0 +1,271 @@
+	public static final String COMMA = ", ";
+	public static final String DOT = ".";
+	public static final String EXISTENTIAL_IDENTIFIER = "!";
+	public static final String UNIVERSAL_IDENTIFIER = "?";
+	public static final String NAMEDNULL_IDENTIFIER = "_";
+	public static final String OPEN_PARENTHESIS = "(";
+	public static final String CLOSING_PARENTHESIS = ")";
+	public static final String RULE_SEPARATOR = " :- ";
+	public static final String AT = "@";
+	public static final String SOURCE = "@source ";
+	public static final String COLON = ": ";
+	public static final String CARET = "^";
+	public static final String LESS_THAN = "<";
+	public static final String MORE_THAN = ">";
+	public static final String ESCAPED_QUOTE = "\"";
+
+	/**
+	 * Constructor.
+	 */
+	private Serializer() {
+
+	}
+
+	private static String escape(String string) {
+		return string.replace("\\", "\\\\").replace("\"", "\\\"");
+	}
+
+	/**
+	 * Creates a String representation of a given {@link Rule}.
+	 * 
+	 * @see <"https://github.com/knowsys/vlog4j/wiki"> for wiki.
+	 * @param rule a {@link Rule}.
+	 * @return String representation corresponding to a given {@link Rule}.
+	 * 
+	 */
+	public static String getString(Rule rule) {
+		return getString(rule.getHead()) + RULE_SEPARATOR + getString(rule.getBody()) + DOT;
+	}
+
+	/**
+	 * Creates a String representation of a given {@link Literal}.
+	 * 
+	 * @see <"https://github.com/knowsys/vlog4j/wiki"> for wiki.
+	 * @param literal a {@link Literal}
+	 * @return String representation corresponding to a given {@link Literal}.
+	 */
+	public static String getString(Literal literal) {
+		final StringBuilder stringBuilder = new StringBuilder("");
+		if (literal.isNegated()) {
+			stringBuilder.append(NEGATIVE_IDENTIFIER);
+		}
+		stringBuilder.append(literal.getPredicate().getName()).append(OPEN_PARENTHESIS);
+		boolean first = true;
+		for (final Term term : literal.getArguments()) {
+			if (first) {
+				first = false;
+			} else {
+				stringBuilder.append(COMMA);
+			}
+			stringBuilder.append(term.getSyntacticRepresentation());
+		}
+		stringBuilder.append(CLOSING_PARENTHESIS);
+		return stringBuilder.toString();
+	}
+
+	/**
+	 * Creates a String representation of a given {@link Fact}.
+	 * 
+	 * @see <"https://github.com/knowsys/vlog4j/wiki"> for wiki.
+	 * @param fact a {@link Fact}
+	 * @return String representation corresponding to a given {@link Fact}.
+	 */
+	public static String getFactString(Fact fact) {
+		return getString(fact) + DOT;
+	}
+
+	/**
+	 * Creates a String representation of a given {@link Constant}.
+	 * 
+	 * @see <"https://github.com/knowsys/vlog4j/wiki"> for wiki.
+	 * @param constant a {@link Constant}
+	 * @return String representation corresponding to a given {@link Constant}.
+	 */
+	public static String getString(Constant constant) {
+		return constant.getName();
+	}
+
+	public static String getString(DatatypeConstant constant) {
+		return getShortConstantName(constant);
```

Comment:
The problem is that the original getConstantName is being used in some other places in code, so changing it will trigger alot of errors, so what do you see ?

Reply:
for example, the reasoner and rdf